xdsl/dialects/arith.py

```diff
@@ -141,29 +141,42 @@ def from_float_and_width(
         return Constant.create(result_types=[typ], attributes={"value": val})
 
 
-class BinaryOperation(IRDLOperation):
+_T = TypeVar("_T")
+
+
+class BinaryOperation(IRDLOperation, Generic[_T]):
     """A generic operation. Operation definitions inherit this class."""
```

Comment:
Could we maybe rename this?

Reply:
@math-fehr I tried something, but I'm not sure how much we want to phrase here. WDYT?